backend/authentication-microservice/src/main/java/com/ss/utopia/auth/service/UserService.java

```diff
@@ -200,4 +169,43 @@ public boolean isUserAdmin(User user) {
   public boolean isUserAgent(User user) {
     return user.getRole().getId() == 3;
   }
+  
+  /**
+   * check whether the user can be saved to the database
+   *
+   * @param user
+   * @return
+   */
+  private User submitUser(User user) {
+	    try {
+	    	return userDao.save(user);
+	    } catch (DataIntegrityViolationException e) {
+	        String error = e.getMessage();
+	        if (error.contains("user.email_UNIQUE")) {
+	          throw new HttpServerErrorException(HttpStatus.CONFLICT, "Email is taken");
+	        } else if (error.contains("user.username_UNIQUE")) {
+	          throw new HttpServerErrorException(HttpStatus.CONFLICT, "Username is taken");
+	        } else if (error.contains("user.phone_UNIQUE")) {
+	          throw new HttpServerErrorException(HttpStatus.CONFLICT, "Phone number is taken");
+	        }
+	    }
+	    return user;
+  }
+  
+  private User updateUserInfo(User user, UpdateUserDto userDto) {
+	  if(userDto.getUsername() != null && userDto.getUsername() != "")
+	    	user.setUsername(userDto.getUsername());
+	    if(userDto.getPassword() != null && userDto.getPassword() != "")
```

Comment:
The formatting here is a bit off

Reply:
When I will try to fix this, but eclipse formatting is thrown off after VSCode formats the page

Reply:
Ignore the typo in my previous comment, I meant to say "I will try ..."